### PR TITLE
Switching to GitHub actions, adding test for Linux, and pdf doc generation

### PR DESCRIPTION
**Pull request by PProfizi**
Needed to switch the CI to GitHub actions.
Took the time to make the Linux testing job work, added the pdf doc generation step.
Prepared for use of pydpf-actions.

### CODE REVIEW

**Comment by PProfizi** [comment]
Don't look at the composite actions too much as they will be moved to pydpf-actions.

**Comment by PProfizi** [comment]
- [ ] as soon as ready to merge, will change the settings for required CI

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,161 +10,53 @@ on:
...
+  Build_and_Test:
+    name: Build and Test
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.8", "3.9"]
-        os: ["windows-2019"]
+        python-version: ["3.7"]
+        os: ["windows-latest", "ubuntu-latest"]
```
Comment (cbellot000) [inline .github/workflows/ci.yml:26]: @PProfizi really cool

File: docs/source/conf.py [hunk 2]
```diff
@@ -117,11 +116,16 @@
 # html_theme_path = [sphinx_rtd_theme.get_html_theme_path()]
 # zerovm-sphinx-theme
 
+html_short_title = html_title = "PyDPF-Post"
 html_theme = "pyansys_sphinx_theme"
-html_logo = "_static/pyansys-logo-black-cropped.png"
+html_logo = pyansys_logo_black
```
Comment (cbellot000) [inline docs/source/conf.py:121]: @PProfizi willl you update it for the core too?

File: docs/source/conf.py [hunk 3]
```diff
@@ -182,7 +186,7 @@
 
 # One entry per manual page. List of tuples
 # (source start file, name, description, authors, manual section).
-man_pages = [(master_doc, "pyansys", "PyAnsys DPF-Post Documentation", [author], 1)]
+man_pages = [(master_doc, "PyDPF-Post", "PyAnsys DPF-Post Documentation", [author], 1)]
```
Comment (cbellot000) [inline docs/source/conf.py:189]: what does that mean?

**Review by cbellot000 — APPROVED** [review]

File: docs/source/conf.py (hunk 2 above)
Comment (PProfizi) [reply docs/source/conf.py:121]: @cbellot000 Yes, it is in https://github.com/pyansys/pydpf-core/pull/234

File: docs/source/conf.py (hunk 3 above)
Comment (PProfizi) [reply docs/source/conf.py:189]: @cbellot000 It is the name of the Tex file created, and thus of the pdf file, giving us a "PyDPF-Post.pdf" file for the manual.

**Pull request merged**